### DISCUSSION TITLE
Xamarin Android foreground service for location tracking is stoped after a few minutes

### DISCUSSION THREAD

**Discussion by MaticDiba** [xamarin/Xamarin.Forms#15407] [Q&A]
Hi,
I'm trying to build an application that users can use on their hikes and track their progress. I prepared an example project: https://github.com/MaticDiba/XFBackgroundLocationSampleExtemded

The issue I'm facing is that after the screen is turned off, the application stops tracking after approximately 2-3 minutes. Then it resumes when I turn on the screen. I guess this is because the phone goes into some kind of sleep/battery save mode.

I tried deploying this project to my phone (Samsung Galaxy S20 - Android 12) via VisualStudio 2022. When I test this application with the screen off, it stops recording location after 2-3 minutes.

I have other tracking applications, and with those, I'm able to do it when the screen is turned off. So I guess the phone is not a problem.

On the other hand, when testing this on Samsung Galaxy S8 (Android 9), it works as it should. 

I also tried acquiring `WakeLock` but with no success. I was using `WakeLockFlags.Partia` lock flag.

**Comment by kunalprakash3891** [comment]
Hi @MaticDiba, this may or may not be related to your case, but we have noticed that some Samsung phones suppress local notifications after a while. They seem to be putting (certain) apps to sleep, preventing any notifications from being delivered. This does not happen on Google Pixel or Apple iPhones, seems to be specific to Samsung phones. We were unable to find a work-around, instead opting to use Firebase for notifications. Perhaps something similar is happening in your scenario.
